rfd/0128-aws-integration-tests.md

```diff
@@ -0,0 +1,191 @@
+- Security: `@reedloden || @jentfoo`
+
+## What
+
+As part of the increase of the reliability of Teleport, we want to add integration
+tests for AWS. This will allow us to test the integration with AWS without having
+to rely on manual testing.
+
+## Why
+
+We want to increase the reliability of Teleport by adding integration tests for
+AWS. This will allow us to test the integration with AWS without having to rely
+on manual testing. This will also allow us to test the integration with AWS in
+a more reliable process to ensure we don't introduce regressions in the future
+or if we do, we can catch them early.
+
+Teleport integrates deeply with AWS to provide a seamless experience for users
+when using auto-discovery of nodes, databases and Kubernetes clusters. This
+integration is critical for the success of Teleport and we want to ensure that
+we can test it reliably.
+
+Each integration test will use the minimum required
+AWS API permissions to test the integration. This will ensure that we don't
+introduce regressions by changing the permissions required by Teleport but those
+changes are not detected because another test requires the same permissions.
+
+The goal of this RFD is to define the scope of the integration tests for AWS
+and how we will run them.
+
+## When
+
+AWS integration tests will be added to the Teleport CI pipeline as part of the
+cronjob that triggers integrations each day. During the release process, the
+integration tests will be run automatically as part of the release process.
+
+## How
+
+This section describes how the end-to-end tests will authenticate with AWS API
+and how they will be authorized to perform the required actions.
+
+Teleport AWS account will be configured to allow GitHub OIDC provider to assume
+a set of roles. These roles will be assumed by the GitHub actions pipeline to
+interact with AWS API.
+
+GitHub action [configure-aws-credentials](https://github.com/aws-actions/configure-aws-credentials)
+will be used to handle the authentication with AWS API. This action configures
+AWS credentials and region environment variables for use in subsequent steps.
+The action implements the AWS SDK credential resolution chain and exports the
+following environment variables:
+
+- `AWS_ACCESS_KEY_ID`
+- `AWS_SECRET_ACCESS_KEY`
+- `AWS_SESSION_TOKEN`
+
+Once the environment variables are set, the AWS SDK that Teleport uses will
+automatically use them to authenticate with the AWS API and does not require
+any additional configuration.
+
+One of the requirements for the integration tests is that they should use the
+minimum required permissions to perform the required actions. This will ensure
+that we don't introduce regressions by changing the permissions required by
+Teleport. To achieve this, GitHub action will be allowed to assume a set of
+roles that will simulate the minimum required permissions that each Teleport
+service requires to perform the required actions. Teleport services will be
```

Comment:
Does Kube Service/Kube Agent support STS Assume role call in the EKS kube access flow? 

I think that this is the kind of feature that is not scoped only to integration tests flow and can be used in case of one Kube Service  - multiple EKS AWS Account access so it is worth to handle/track this feature in a separate issue.

Reply:
Kubernetes service will support assuming role as part of the matcher selector

I imagined something like:

```yaml
kubernetes_service:
  enabled: "yes"
  resources:
  - tags:
      "env": "prod"
    assume_aws_role: ...
```